src/Functions/parseDateTime.cpp

```diff
@@ -672,7 +734,7 @@ namespace
                 Int64OrError result = 0;
 
                 /// Ensure all input was consumed
-                if (cur < end)
+                if (!parseDateTime64 && cur < end)
```

Comment:
It is not clear why we no longer check for `parseDateTime64`. Please add a comment.